src/analysis/souffle/authorization_logic.dl

```diff
@@ -52,12 +52,33 @@ isPrincipal(speaker) :- saysDowngrades(speaker, _, _).
 //-----------------------------------------------------------------------------
 downgrades(path, tag) :- ownsTag(owner, tag), saysDowngrades(owner, path, tag).
 
+// This rule conservatively says that if _any_ principal claims that a path has
+// a tag, then it has the tag according to the taint analysis. This rule is
+// likely to be adjusted later. For confidentiality tags this should be OK to
+// start with.
+claimHasTag(path, tag) :- ownsTag(owner, tag), saysHasTag(_, path, tag).
+
+// This rule says that any principal can say that another principal owns a tag.
+// This rule is just a starting point to allow all aspects of the policy to
+// come from authorization logic -- it is just a simple way to generate
+// ownsTag from something with a speaker. This rule is likely to change. It
+// does not rule out the case that two different principals claim ownership of
+// a tag.
+ownsTag(speaker, tag) :- saysOwnsTag(speaker, owner, tag).
```

Comment:
Is this right? I would think that "any speaker may say that any principal may own a tag" would be written as

```
ownsTag(owner, tag) :- isPrincipal(speaker), saysOwnsTag(speaker, owner, tag).

Reply:
The key point of my comment isn't about the `isPrincipal` part. What I'm saying is that your comment says that:

```
This rule says that any principal can say that another principal owns a tag.
```

My interpretation of the rule you wrote in English is "A `speaker` owns a `tag` if `speaker` says any `owner` owns that `tag`". I may be misunderstanding, but the best way to match the comment to the rule would be Datalog that instead says "An `owner` owns a `tag if any `speaker` says that `owner` owns that `tag`". An equivalent way to write my suggestion is:

```
ownsTag(owner, tag) :- saysOwnsTag(_, owner, tag).
```

Reply:
Oh yes. The rule you wrote there is what I actually meant to put. I overindexed on the `isPrincipal` part. Thanks and good catch.